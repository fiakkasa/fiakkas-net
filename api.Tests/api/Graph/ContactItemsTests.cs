using HotChocolate.Execution;

namespace api.Tests;

public class ContactItemsTests(GraphFixture fixture) : IClassFixture<GraphFixture>
{
    [Fact]
    public async Task ContactItems_Should_Return_Data()
    {
        var executor = await fixture.GetRequestExecutor();

        var result = await executor.ExecuteAsync(
"""
{
  contactItems {
    totalCount
    items {
      createdAt
      description
      href
      icon
      id
      key
      title
      updatedAt
      version
    }
  }
}
""");

        Func<IQueryResult> fn = result.ExpectQueryResult;
        fn.Should().NotThrow();
        fn().Errors.Should().BeNullOrEmpty();
        result.ToJson().MatchSnapshot();
    }
}
