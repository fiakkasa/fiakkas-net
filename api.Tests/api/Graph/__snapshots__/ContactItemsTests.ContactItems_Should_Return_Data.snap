{
  "data": {
    "contactItems": {
      "totalCount": 1,
      "items": [
        {
          "content": "Content",
          "createdAt": "2024-01-01T00:00:00.000Z",
          "href": "/test",
          "icon": "Icon",
          "id": "ebf224a8-7ff3-47b9-882b-dd41ec7f5a05",
          "key": "Key",
          "title": "Title",
          "updatedAt": null,
          "version": 1
        }
      ]
    }
  }
}
